test/TestUtilities/Test.Utility/TestServer/PortReserver.cs

```diff
@@ -57,21 +59,26 @@ public PortReserver(int basePort = 50231)
 
                 // WaitForLockAsync prevents port contention with this app.
                 string portLockName = $"NuGet-Port-{port}";
-                var tryOnceCts = new CancellationTokenSource(TimeSpan.Zero);
+
                 try
                 {
-                    var attemptedPort = port;
-                    return await ConcurrencyUtilities.ExecuteWithFileLockedAsync<T>(
-                        portLockName,
-                        t => action(attemptedPort, token),
-                        tryOnceCts.Token);
+                    if (PortLock.TryAdd(portLockName, true))
```

Comment:
Can you elaborate more on the solution? 
You are changing from a machine wide make shift locking system to a process only one.

Reply:
It's a test helper. We don't run multiple test processes at the same time, so I don't think it should be a problem.

Reply:
I get that...I'm just not sure what we are fixing here :)
It seems like the timespan is the problem, but that wasn't called out.

Reply:
It seems that before net 5, the cancellation token was being created without yet requesting cancellation, even if `TimeSpan.Zero` is used, and then later on (async continuations?) it starts requesting cancellation. It appears that either this cancellation is faster in net5, or the token is created in a cancelled state. In any case, Heng told me it always fails on netcoreapp5.0, but still works on net472 and netcoreapp2.1.

`ConcurrencyUtilities` previously guaranteed entry when nothing already had a lock on the file, even if the cancellation token was already requesting cancellation. When I added `KeyedLock` I removed that guarantee.

Reply:
That makes sense, thanks.

Reply:
Thanks! Updated in the descriptions.